apiserver/facades/client/modelmanager/listmodelsinfo_test.go

```diff
@@ -0,0 +1,177 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package modelmanager_test
+
+import (
+	"github.com/juju/errors"
+	gitjujutesting "github.com/juju/testing"
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/juju/apiserver/facades/client/modelmanager"
+	"github.com/juju/juju/apiserver/params"
+	apiservertesting "github.com/juju/juju/apiserver/testing"
+	"github.com/juju/juju/environs/config"
+	"github.com/juju/juju/permission"
+	_ "github.com/juju/juju/provider/azure"
+	"github.com/juju/juju/provider/dummy"
+	_ "github.com/juju/juju/provider/ec2"
+	_ "github.com/juju/juju/provider/joyent"
+	_ "github.com/juju/juju/provider/maas"
+	_ "github.com/juju/juju/provider/openstack"
+	"github.com/juju/juju/state"
+	jujuversion "github.com/juju/juju/version"
+)
+
+type ListModelsWithInfoSuite struct {
+	gitjujutesting.IsolationSuite
+
+	st *mockState
+
+	authoriser apiservertesting.FakeAuthorizer
+	adminUser  names.UserTag
+
+	api *modelmanager.ModelManagerAPI
+}
+
+var _ = gc.Suite(&ListModelsWithInfoSuite{})
+
+func (s *ListModelsWithInfoSuite) SetUpTest(c *gc.C) {
+	s.IsolationSuite.SetUpTest(c)
+
+	adminUser := "admin"
+	s.adminUser = names.NewUserTag(adminUser)
+
+	s.st = &mockState{
+		model: s.createModel(c, s.adminUser),
+	}
+	s.st.modelDetailsForUser = func() ([]state.ModelDetails, error) {
+		return []state.ModelDetails{s.st.model.getModelDetails()}, s.st.NextErr()
+	}
+
+	s.authoriser = apiservertesting.FakeAuthorizer{
+		Tag: s.adminUser,
+	}
+	api, err := modelmanager.NewModelManagerAPI(s.st, &mockState{}, nil, s.authoriser, s.st.model)
+	c.Assert(err, jc.ErrorIsNil)
+	s.api = api
+}
+
+func (s *ListModelsWithInfoSuite) createModel(c *gc.C, user names.UserTag) *mockModel {
+	attrs := dummy.SampleConfig()
+	attrs["agent-version"] = jujuversion.Current.String()
+	cfg, err := config.New(config.UseDefaults, attrs)
+	c.Assert(err, jc.ErrorIsNil)
+
+	return &mockModel{
+		owner: user,
+		cfg:   cfg,
+	}
+}
+
+func (s *ListModelsWithInfoSuite) setAPIUser(c *gc.C, user names.UserTag) {
+	s.authoriser.Tag = user
+	modelmanager, err := modelmanager.NewModelManagerAPI(s.st, &mockState{}, nil, s.authoriser, s.st.model)
+	c.Assert(err, jc.ErrorIsNil)
+	s.api = modelmanager
+}
+
+func (s *ListModelsWithInfoSuite) TestListModelsWithInfoForSelf(c *gc.C) {
+	user := names.NewUserTag("external@remote")
```

Comment:
Do we have another model that this is filtering out? Or are we only asserting that external users can see models?
Should we have 2 user tags here and see that we see one but not the other?
Should we be confirming that we are seeing the *correct* one, and not just the count?
maybe a helper for "checkCorrectModelsReturned(result, []string{s.st.model.Name})" ?
This looks a lot like we have a single model and if we didn't do any validation of scope we would pass this test.

Given the implementation of:
s.st.modelDetailsForUser = func() ([]state.ModelDetails, error) {
+		return []state.ModelDetails{s.st.model.getModelDetails()}, s.st.NextErr()
+	}

Is it *possible* for us to ever return > 1 model?
I'm trying to figure out what we're actually testing. Maybe the local vs remote distinction doesn't mean anything, and we have a test but its intentionally small.